sigoptlite/sigoptlitetest/endpoints/test_observation.py

```diff
@@ -9,70 +12,493 @@
+  ):
+    with pytest.raises(SigOptException):
+      connection.experiments(experiment.id).observations().create(
+        suggestion=suggestion.id,
+        **failed_params,
+        **values_params,
+      )
+
+  def test_multiple_observations(self, connection):
+    num_observations = 7
+    experiment_meta = self.get_experiment_feature("simple")
+
+    e = connection.experiments().create(**experiment_meta)
+    for i in range(num_observations):
+      observation = (
+        connection.experiments(e.id)
+        .observations()
+        .create(
+          assignments={"d1": 0.5 + i, "i1": 10 + i},
+          values=[{"name": "y1", "value": i}],
+        )
+      )
+      assert observation.assignments["d1"] == 0.5 + i
+      assert observation.assignments["i1"] == 10 + i
+      assert observation.values[0].value == i
+
+    observations = connection.experiments(e.id).observations().fetch()
+    assert observations.count == num_observations
+    assert len(observations.data) == num_observations
+    for i, obs in enumerate(observations.data):
+      assert obs.assignments["d1"] == 0.5 + i
+      assert obs.assignments["i1"] == 10 + i
+      assert obs.values[0].value == i
+
+  @pytest.mark.parametrize(
+    "feature",
+    [
+      "constraints",
+      "conditionals",
+      "default",
+      "multisolution",
+      "priors",
+    ],
+  )
+  def test_single_metric_observations(self, connection, feature):
+    num_observations = 4
+    experiment_meta = self.get_experiment_feature(feature)
+
+    e = connection.experiments().create(**experiment_meta)
+    assert len(e.metrics) == 1
+    metric_name = e.metrics[0].name
+
+    assignments_list = []
+    values_list = []
+    for i in range(num_observations):
+      suggestion = connection.experiments(e.id).suggestions().create()
+      assignments = dict(suggestion.assignments)
+      values = [{"name": metric_name, "value": i}]
+      observation = self.create_observation_endpoint(
+        connection=connection,
+        experiment=e,
+        assignments=assignments,
+        values=values,
+        suggestion_id=suggestion.id,
+        use_suggestion_id=i % 2,
+      )
+      for p in assignments.keys():
+        assert observation.assignments[p] == assignments[p]
+      assert len(observation.values) == 1
+      assert observation.values[0].name == metric_name
+      assert observation.values[0].value == i
+      assignments_list.append(assignments)
+      values_list.append(values)
+
+    observations = connection.experiments(e.id).observations().fetch()
+    assert observations.count == num_observations
+    assert len(observations.data) == num_observations
+    for obs, params, vals in zip(observations.data, assignments_list, values_list):
+      for p in params.keys():
+        assert obs.assignments[p] == params[p]
+      assert obs.values[0].name == vals[0]["name"]
+      assert obs.values[0].value == vals[0]["value"]
```

Comment:
You could do this inside the loop (starting at line 303) and fetch the observations at each iteration, then use `next()` to get the most recent observation and check that it has the correct assignments and values.

Reply:
that's not the goal of this test. we want to create observations, and then loop over the fetched observations